File: pepper-farm/frontend/src/types/chatbot.ts
/**
 * Types for the visitor AI chatbot.
 */

/** A single message in the chat conversation, used by the UI. */
export interface ChatMessage {
  role: "user" | "assistant";
  content: string;
}

/**
 * The backend reply from POST /api/chatbot.
 * `source` indicates where the answer came from:
 *   - "db"       → grounded in internal data from the database
 *   - "ai"       → a general recommendation from OpenAI
 *   - "fallback" → no matching data / service unavailable
 */
export interface ChatResponse {
  answer: string;
  source: "db" | "ai" | "fallback";
}

File: pepper-farm/backend/routers/chatbot.py
"""
Chatbot router.

Exposes a single PUBLIC endpoint for the visitor AI chatbot:
    POST /api/chatbot

No authentication is required — visitors are not logged in. A read-only DB
session is injected via Depends(get_db) and passed to the service so the
service can fetch grounding facts. The endpoint is read-only and the service
never raises, so this endpoint always returns 200 with a ChatResponse (a safe
fallback reply when OpenAI is unavailable or when a factual question has no
matching DB data).
"""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from database import get_db
from schemas.chatbot import ChatRequest, ChatResponse
from services.chatbot_service import answer_question

router = APIRouter(prefix="/api/chatbot", tags=["chatbot"])


@router.post("", response_model=ChatResponse)
def chat(request: ChatRequest, db: Session = Depends(get_db)) -> ChatResponse:
    """Answer a visitor's question via the chatbot service."""
    return answer_question(db, request.message)
